crates/oxc_semantic/src/reference.rs

```diff
@@ -36,28 +39,47 @@ impl Reference {
         self.symbol_id = Some(symbol_id);
     }
 
+    /// Returns `true` if the identifier value was read. This is not mutually
+    /// exclusive with [`#is_write`]
     pub fn is_read(&self) -> bool {
-        self.flag == ReferenceFlag::Read
+        self.flag.is_read()
     }
 
+    /// Returns `true` if the identifier was written to. This is not mutually
+    /// exclusive with [`#is_read`]
     pub fn is_write(&self) -> bool {
-        self.flag == ReferenceFlag::Write
+        self.flag.is_write()
     }
 }
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-pub enum ReferenceFlag {
-    None,
-    Read,
-    Write,
+bitflags! {
```

Comment:
out of curiousity, what purpose does making this a bitflags! serve?

Reply:
It allows for unioning read/write flags